third-party/realdds/doc/DDS ICD.md

```diff
@@ -0,0 +1,147 @@
+
+When using librealsense API, received notifications will automatically be handled and reflected in the `rs2::device` and `rs2::sensor`s state.
+
+All notifications use the **flexible** type with JSON format
+
+### Discovery notifications
+
+On the discovery phase the camera server will send a **device-header** message, a possible **device-options** message and for each stream a **stream-header** and **stream-options** messages.
+
+#### device-header format example
+
+    "id":"device-header",
+    "n-streams": 5        //Integer, number of streams
+
+#### device-options format example
+
+    "id":"device-options",
+    "n-options": 2,       //Integer, number of device options
+    "options":            //Array of supported options, the size of n-options
+    [
+        {"description":"Enable/Disable global timestamp","name":"Global Time Enabled","range-default":1.0,"range-max":1.0,"range-min":0.0,"range-step":1.0,"value":1.0},
+        {"description":"HDR Option","name":"Hdr Enabled","range-default":0.0,"range-max":1.0,"range-min":0.0,"range-step":1.0,"value":0.0}
+    ]
+
+#### stream-header format example
+
+    "default-profile-index":0,
+    "id":"stream-header",
+    "name":"Infrared 1",
+    "profiles":           //All stream profiles. Each profile is an array of frequency, format, width, height.
+    [
+        [25,"Y16",1288,808],[15,"Y16",1288,808],[30,"GREY",1280,720],[15,"GREY",1280,720],[5,"GREY",1280,720],[90,"GREY",848,480],[60,"GREY",848,480],[30,"GREY",848,480],[15,"GREY",848,480],[5,"GREY",848,480],[90,"GREY",640,480],[60,"GREY",640,480],[30,"GREY",640,480],[15,"GREY",640,480],[5,"GREY",640,480],[90,"GREY",640,360],[60,"GREY",640,360],[30,"GREY",640,360],[15,"GREY",640,360],[5,"GREY",640,360],[90,"GREY",480,270],[60,"GREY",480,270],[30,"GREY",480,270],[15,"GREY",480,270],[5,"GREY",480,270],[90,"GREY",424,240],[60,"GREY",424,240],[30,"GREY",424,240],[15,"GREY",424,240],[5,"GREY",424,240]
+    ],
+    "sensor-name":"Stereo Module",
+    "type":"ir"
+
+#### stream-options format example
+
+    "id":"stream-options",
+    "n-options":3,        //Integer, number of device options
+    "options":            //Array of supported options, the size of n-options
+    [
+        {"description":"Enable / disable backlight compensation","name":"Backlight Compensation","range-default":0.0,"range-max":1.0,"range-min":0.0,"range-step":1.0,"value":0.0},
+        {"description":"UVC image brightness","name":"Brightness","range-default":0.0,"range-max":64.0,"range-min":-64.0,"range-step":1.0,"value":0.0},
+        {"description":"UVC image contrast","name":"Contrast","range-default":50.0,"range-max":100.0,"range-min":0.0,"range-step":1.0,"value":50.0}
+    ],
+    "stream-name":"Infrared 1"
+
+### Instant notifications
+
+Other notifications are sent as needed, as a reply to **set-option** or **query-option**
+
+#### query-option reply format example
```

Comment:
short code for executing a query-option operation would be great to have here

Reply:
Code examples are out of scope for this document, they can be added later in another document